src/auth/googleauth.ts

```diff
@@ -203,23 +203,32 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
     // - Cloud SDK: `gcloud config config-helper --format json`
     // - GCE project ID from metadata server)
     if (!this._getDefaultProjectIdPromise) {
-      this._getDefaultProjectIdPromise = (async () => {
-        const projectId =
-          this.getProductionProjectId() ||
-          (await this.getFileProjectId()) ||
-          (await this.getDefaultServiceProjectId()) ||
-          (await this.getGCEProjectId()) ||
-          (await this.getExternalAccountClientProjectId());
-        this._cachedProjectId = projectId;
-        if (!projectId) {
-          throw new Error(
-            'Unable to detect a Project Id in the current environment. \n' +
-              'To learn more about authentication and Google APIs, visit: \n' +
-              'https://cloud.google.com/docs/authentication/getting-started'
-          );
+      // TODO: refactor the below code so that it doesn't mix and match
+      // promises and async/await.
+      this._getDefaultProjectIdPromise = new Promise(
+        // eslint-disable-next-line no-async-promise-executor
+        async (resolve, reject) => {
+          try {
+            const projectId =
+              this.getProductionProjectId() ||
+              (await this.getFileProjectId()) ||
+              (await this.getDefaultServiceProjectId()) ||
+              (await this.getGCEProjectId()) ||
+              (await this.getExternalAccountClientProjectId());
+            this._cachedProjectId = projectId;
+            if (!projectId) {
+              throw new Error(
+                'Unable to detect a Project Id in the current environment. \n' +
+                  'To learn more about authentication and Google APIs, visit: \n' +
+                  'https://cloud.google.com/docs/authentication/getting-started'
+              );
+            }
+            resolve(projectId);
+          } catch (e) {
+            reject(e);
+          }
```

Comment:
From what I can understand from the context of this PR, it seems to me that the goal of the proposed changes here is for the `this._getDefaultProjectIdPromise` variable to hold a `Promise<projectId>` value instead of directly holding the `projectId` value. With that in mind I believe this change would translate that intent, while allowing to get rid of the unwanted mix and match of promises/async/await:

```suggestion
      this._getDefaultProjectIdPromise =
        Promise.resolve(this.getProductionProjectId())
          .then(projectId => projectId ?? this.getFileProjectId())
          .then(projectId => projectId ?? this.getDefaultServiceProjectId())
          .then(projectId => projectId ?? this.getGCEProjectId())
          .then(projectId => projectId ?? this.getExternalAccountClientProjectId())
          .then(projectId => {
            if (projectId) {
              this._cachedProjectId = projectId;
              return projectId
            } else {
              throw new Error(
                'Unable to detect a Project Id in the current environment. \n' +
                  'To learn more about authentication and Google APIs, visit: \n' +
                  'https://cloud.google.com/docs/authentication/getting-started'
              );
            }
          })

```